ts/screens/profile/PreferencesScreen.tsx

```diff
@@ -85,18 +85,7 @@ class PreferencesScreen extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props);
     this.state = INITIAL_STATE;
-    this.handleEmailOnPress = this.handleEmailOnPress.bind(this);
-  }
 
-  private handleEmailOnPress() {
-    if (this.props.hasProfileEmail) {
-      this.props.navigateToEmailReadScreen();
-      return;
-    }
-    this.props.navigateToEmailInsertScreen();
-  }
-
-  public componentWillMount() {
```

Comment:
I would prefer to keep this logic inside the componentWillMount (or componentDidMount) because it affects UI stuffs
see https://stackoverflow.com/questions/34961853/what-will-happen-if-i-use-setstate-function-in-constructor-of-a-class-in-react
What do you think?